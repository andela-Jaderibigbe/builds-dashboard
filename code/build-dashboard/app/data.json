[
  {
    "_id": 1,
    "user": "",
    "build_name": "Tenrox-R1_1235",
    "state": "pending",
    "progress": {
      "metrics": 0,
      "build": 0,
      "unit_test": 0,
      "functional_test": 0
    },
    "stats": {
      "metrics": null,
      "build": null,
      "unit_test": null,
      "functional_test": null
    },
    "time_started": null
  },
  {
    "_id": 2,
    "user": "jack",
    "build_name": "432462",
    "state": "running",
    "progress": {
      "metrics": 100,
      "build": 20,
      "unit_test": 0,
      "functional_test": 0
    },
    "stats": {
      "metrics": {
        "test": 64,
        "maintainability": 53,
        "security": 64,
        "workmanship": 72
      },
      "build": null,
      "unit_test": null,
      "functional_test": null
    },
    "time_started": "2017-10-17T15:48:12.600Z"
  },
  {
    "_id": 3,
    "user": "samy",
    "build_name": "432461",
    "state": "rejected",
    "progress": {
      "metrics": 100,
      "build": 100,
      "unit_test": 100,
      "functional_test": 100
    },
    "stats": {
      "metrics": {
        "test": 64,
        "maintainability": -53,
        "security": 64,
        "workmanship": 72
      },
      "build": {
        "build_date": "2017-10-15T10:48:12.600Z"
      },
      "unit_test": {
        "coverage": 0.76,
        "test_passing": 0.73,
        "covered": 142,
        "missed": 10
      },
      "functional_test": {
        "coverage": 0.23,
        "test_passing": 0.68,
        "covered": 4321,
        "missed": 2145
      }
    },
    "time_started": "2017-10-13T13:48:12.600Z"
  },
  {
    "_id": 4,
    "user": "",
    "build_name": "Tenrox-R1_1234",
    "state": "complete",
    "progress": {
      "metrics": 100,
      "build": 100,
      "unit_test": 100,
      "functional_test": 100
    },
    "stats": {
      "metrics": {
        "test": 64,
        "maintainability": 53,
        "security": 64,
        "workmanship": 72
      },
      "build": {
        "build_date": "2017-10-12T14:48:12.600Z"
      },
      "unit_test": {
        "coverage": 0.76,
        "test_passing": 0.73,
        "covered": 142,
        "missed": 10
      },
      "functional_test": {
        "coverage": 0.23,
        "test_passing": 0.68,
        "covered": 4321,
        "missed": 2145
      }
    },
    "time_started": "2017-10-11T11:48:12.600Z"
  },
  {
    "_id": 5,
    "user": "samy",
    "build_name": "432460",
    "state": "rejected",
    "progress": {
      "metrics": 100,
      "build": 0,
      "unit_test": 0,
      "functional_test": 0
    },
    "stats": {
      "metrics": {
        "test": 64,
        "maintainability": -53,
        "security": 64,
        "workmanship": 72
      },
      "build": null,
      "unit_test": null,
      "functional_test": null
    },
    "time_started": "2017-07-11T15:48:12.600Z"
  },
  {
    "_id": 6,
    "user": "samy",
    "build_name": "432459",
    "state": "accepted",
    "progress": {
      "metrics": 100,
      "build": 100,
      "unit_test": 100,
      "functional_test": 100
    },
    "stats": {
      "metrics": {
        "test": 64,
        "maintainability": 53,
        "security": 64,
        "workmanship": 72
      },
      "build": {
        "build_date": "2017-09-13T18:48:12.600Z"
      },
      "unit_test": {
        "coverage": 0.76,
        "test_passing": 0.73,
        "covered": 142,
        "missed": 10
      },
      "functional_test": {
        "coverage": 0.23,
        "test_passing": 0.68,
        "covered": 4321,
        "missed": 2145
      }
    },
    "time_started": "2017-08-13T12:48:12.600Z"
  }
]
