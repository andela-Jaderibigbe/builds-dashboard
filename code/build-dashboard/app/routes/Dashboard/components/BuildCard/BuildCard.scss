@import 'theme/variables';

.labelText {
  font-size: 17px;
}

.stat {
  padding: 15px 0;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  color: $blue;
}

.statItem {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  i {
    font-size: 60px;
  }
}

.bottomLabel {
  color: $black;
  text-align: center;
  font-weight: 500;
}
