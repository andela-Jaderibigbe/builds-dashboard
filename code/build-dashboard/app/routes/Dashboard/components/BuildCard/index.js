import React from 'react';
import PropTypes from 'prop-types';
import Card from 'components/Card';
import { formatBuildDate } from 'utils';
import styles from './BuildCard.scss';

const BuildCard = props => {
  const { state, progress, hasError, build_date, hasContent } = props;

  const renderContent = (
    <span>
      <div className={styles.stat}>
        <span className={styles.statItem}>
          <i className="fa fa-laptop" />
          <span className={styles.labelText}>Debug</span>
        </span>
        <span className={styles.statItem}>
          <i className="fa fa-laptop" />
          <span className={styles.labelText}>Release</span>
        </span>
      </div>
      <div className={styles.bottomLabel}>{formatBuildDate(build_date)}</div>
    </span>
  );

  return (
    <Card
      title="Build"
      state={state}
      progress={progress}
      hasError={hasError}
      hasContent={hasContent}
    >
      {renderContent}
    </Card>
  );
};

BuildCard.propTypes = {
  build_date: PropTypes.string,
  state: PropTypes.string,
  progress: PropTypes.number,
  hasError: PropTypes.bool,
  hasContent: PropTypes.bool,
};

export default BuildCard;
