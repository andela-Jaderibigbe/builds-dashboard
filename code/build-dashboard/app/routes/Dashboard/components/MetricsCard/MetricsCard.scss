@import 'theme/variables';

.cardSection {
  display: flex;
  justify-content: space-around;
  margin-bottom: 10px;
}

.statItem {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.stats {
  font-size: 35px;

  i {
    margin-left: 5px;
  }
}

.labelText {
  color: $black;
  font-weight: 500;
}

.valueDrop {
  color: $red;
}

.valueStable {
  color: $green;
}

.bottomStat {
  @extend .labelText;

  i {
    color: orange;
  }
}
