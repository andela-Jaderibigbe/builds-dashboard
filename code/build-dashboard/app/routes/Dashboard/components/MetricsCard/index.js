import React from 'react';
import PropTypes from 'prop-types';
import Card from 'components/Card';
import { valueDrop } from '../../selector';
import styles from './MetricsCard.scss';

const MetricsCard = props => {
  const {
    state,
    progress,
    hasError,
    test,
    maintainability,
    security,
    workmanship,
    hasContent,
  } = props;

  const renderStat = (value, stableClass, dropClass, stableIcon, dropIcon) => {
    const hasError = valueDrop(value);
    return (
      <span className={`${styles.stats} ${hasError ? dropClass : stableClass}`}>
        {Math.abs(value)}
        {hasError ? <i className={`fa ${dropIcon}`} /> : <i className={`fa ${stableIcon}`} />}
      </span>
    );
  };

  const renderContent = (
    <span>
      <div className={styles.cardSection}>
        <div className={styles.statItem}>
          {renderStat(
            test,
            styles.valueStable,
            styles.valueDrop,
            'fa-arrow-circle-up',
            'fa-arrow-circle-down'
          )}
          <span className={styles.labelText}>Test</span>
        </div>
        <div className={styles.statItem}>
          {renderStat(
            maintainability,
            styles.valueStable,
            styles.valueDrop,
            'fa-arrow-circle-up',
            'fa-arrow-circle-down'
          )}
          <span className={styles.labelText}>Maintainability</span>
        </div>
      </div>
      <div className={styles.cardSection}>
        <div className={styles.statItem}>
          {renderStat(
            security,
            styles.bottomStat,
            styles.valueDrop,
            'fa-arrow-circle-right',
            'fa-arrow-circle-left'
          )}
          <span className={styles.labelText}>Security</span>
        </div>
        <div className={styles.statItem}>
          {renderStat(
            workmanship,
            styles.bottomStat,
            styles.valueDrop,
            'fa-arrow-circle-right',
            'fa-arrow-circle-left'
          )}
          <span className={styles.labelText}>Workmanship</span>
        </div>
      </div>
    </span>
  );

  return (
    <Card
      title="Metrics"
      state={state}
      progress={progress}
      hasError={hasError}
      hasContent={hasContent}
    >
      {renderContent}
    </Card>
  );
};

MetricsCard.propTypes = {
  test: PropTypes.number,
  maintainability: PropTypes.number,
  security: PropTypes.number,
  workmanship: PropTypes.number,
  state: PropTypes.string,
  progress: PropTypes.number,
  hasError: PropTypes.bool,
  hasContent: PropTypes.bool,
};

export default MetricsCard;
