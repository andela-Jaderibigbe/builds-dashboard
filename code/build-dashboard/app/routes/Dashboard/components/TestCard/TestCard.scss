@import 'theme/variables';

$lightgreen: #c8dfb8;
$lightorange: #f5ccb1;

.coverageContainer {
  height: 30px;
  background-color: $lightorange;
  width: 95%;
  margin: 0 auto;
  position: relative;
}

.coverageCovered {
  background-color: $lightgreen;
  display: block;
  height: inherit;
  position: absolute;
}

.summary {
  position: absolute;
  width: 100%;
  height: inherit;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.statsText {
  font-size: 15px;
  font-weight: 500;
  color: $black;
}

.info {
  font-size: 10px;
  color: $black;
}

.chartStatsContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  margin-bottom: 10px;
}

.totalTests {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.titleText {
  font-size: 35px;
}

.pieChartContainer {
  width: 45%;
}
