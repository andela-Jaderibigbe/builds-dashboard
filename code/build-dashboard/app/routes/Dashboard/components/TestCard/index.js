import React from 'react';
import { PieChart } from 'react-easy-chart';
import PropTypes from 'prop-types';
import Card from 'components/Card';
import { formatCoverage } from 'utils';

import styles from './TestCard.scss';

const TestCard = props => {
  const {
    title,
    state,
    progress,
    hasError,
    coverage,
    covered,
    missed,
    test_passing,
    hasContent,
  } = props;

  const renderContent = () => (
    <span>
      <div className={styles.chartStatsContainer}>
        <div className={styles.pieChartContainer}>
          <PieChart
            labels
            size={100}
            data={[
              { key: `${covered}`, color: '#43A19E', value: covered },
              { key: `${missed}`, color: '#FF9824', value: missed },
            ]}
            styles={{
              '.chart_text': {
                fontSize: '0.8em',
                fill: '#fff',
              },
            }}
          />
        </div>
        <div className={styles.totalTests}>
          <span className={styles.titleText}>{formatCoverage(test_passing)}</span>
          <span className={styles.info}>tests passed</span>
        </div>
      </div>
      <div className={styles.coverageContainer}>
        <span
          className={styles.coverageCovered}
          style={{ width: formatCoverage(coverage), opacity: 1 }}
        />
        <span className={styles.summary}>
          <span className={styles.statsText}>{formatCoverage(coverage)}</span>
          <span className={styles.info}>code covered</span>
        </span>
      </div>
    </span>
  );

  return (
    <Card
      title={title}
      progress={progress}
      state={state}
      hasError={hasError}
      hasContent={hasContent}
    >
      {renderContent()}
    </Card>
  );
};

TestCard.propTypes = {
  title: PropTypes.string,
  coverage: PropTypes.number,
  test_passing: PropTypes.number,
  covered: PropTypes.number,
  missed: PropTypes.number,
  state: PropTypes.string,
  progress: PropTypes.number,
  hasError: PropTypes.bool,
  hasContent: PropTypes.bool,
};

export default TestCard;
