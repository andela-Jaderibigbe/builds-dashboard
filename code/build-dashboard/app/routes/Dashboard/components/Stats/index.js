import React from 'react';
import PropTypes from 'prop-types';
import MetricsCard from '../MetricsCard';
import BuildCard from '../BuildCard';
import TestCard from '../TestCard';
import ResultCard from '../ResultCard';
import { getError, hasContent } from '../../selector';
import styles from './Stats.scss';

const Stats = props => {
  const { state, progress, stats } = props;

  const { metrics = {}, build = {}, unit_test = {}, functional_test = {} } = stats;

  return (
    <div className={styles.stats}>
      <MetricsCard
        state={state}
        progress={progress.metrics}
        {...metrics}
        hasError={getError(metrics)}
        hasContent={hasContent(progress.metrics)}
      />
      <BuildCard
        state={state}
        progress={progress.build}
        {...build}
        hasError={getError(build)}
        hasContent={hasContent(progress.build)}
      />
      <TestCard
        title="Unit Test"
        {...unit_test}
        state={state}
        progress={progress.unit_test}
        hasError={getError(unit_test)}
        hasContent={hasContent(progress.unit_test)}
      />
      <TestCard
        title="Functional Test"
        {...functional_test}
        state={state}
        progress={progress.functional_test}
        hasError={getError(functional_test)}
        hasContent={hasContent(progress.functional_test)}
      />
      <ResultCard state={state} />
    </div>
  );
};

Stats.propTypes = {
  stats: PropTypes.object,
  state: PropTypes.string,
  progress: PropTypes.object,
};

export default Stats;
