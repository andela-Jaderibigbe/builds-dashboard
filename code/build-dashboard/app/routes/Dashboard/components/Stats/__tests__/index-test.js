import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { expect as chaiExpect } from 'chai';
import data from 'data.json';

import MetricsCard from 'routes/Dashboard/components/MetricsCard';
import BuildCard from 'routes/Dashboard/components/BuildCard';
import TestCard from 'routes/Dashboard/components/TestCard';

import Stats from '../';

describe('<Stats />', () => {
  const props = data[2];
  it('should render correctly', () => {
    const tree = renderer.create(<Stats {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should appropraite cards correctly', () => {
    const wrapper = shallow(<Stats {...props} />);

    chaiExpect(wrapper.find(MetricsCard)).to.have.length(1);
    chaiExpect(wrapper.find(BuildCard)).to.have.length(1);
    chaiExpect(wrapper.find(TestCard)).to.have.length(2);
  });
});
