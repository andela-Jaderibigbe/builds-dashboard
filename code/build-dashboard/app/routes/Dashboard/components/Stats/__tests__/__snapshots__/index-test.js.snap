// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stats /> should render correctly 1`] = `
<div
  className="stats"
>
  <div
    className="cardContainer isError"
    onClick={undefined}
  >
    <div
      className="title"
    >
      Metrics
    </div>
    <div
      className="cardBody"
    >
      <span>
        <div
          className="cardSection"
        >
          <div
            className="statItem"
          >
            <span
              className="stats valueStable"
            >
              64
              <i
                className="fa fa-arrow-circle-up"
              />
            </span>
            <span
              className="labelText"
            >
              Test
            </span>
          </div>
          <div
            className="statItem"
          >
            <span
              className="stats valueDrop"
            >
              53
              <i
                className="fa fa-arrow-circle-down"
              />
            </span>
            <span
              className="labelText"
            >
              Maintainability
            </span>
          </div>
        </div>
        <div
          className="cardSection"
        >
          <div
            className="statItem"
          >
            <span
              className="stats bottomStat"
            >
              64
              <i
                className="fa fa-arrow-circle-right"
              />
            </span>
            <span
              className="labelText"
            >
              Security
            </span>
          </div>
          <div
            className="statItem"
          >
            <span
              className="stats bottomStat"
            >
              72
              <i
                className="fa fa-arrow-circle-right"
              />
            </span>
            <span
              className="labelText"
            >
              Workmanship
            </span>
          </div>
        </div>
      </span>
    </div>
  </div>
  <div
    className="cardContainer isSuccess"
    onClick={undefined}
  >
    <div
      className="title"
    >
      Build
    </div>
    <div
      className="cardBody"
    >
      <span>
        <div
          className="stat"
        >
          <span
            className="statItem"
          >
            <i
              className="fa fa-laptop"
            />
            <span
              className="labelText"
            >
              Debug
            </span>
          </span>
          <span
            className="statItem"
          >
            <i
              className="fa fa-laptop"
            />
            <span
              className="labelText"
            >
              Release
            </span>
          </span>
        </div>
        <div
          className="bottomLabel"
        >
          11:48 am - 10/15/2017
        </div>
      </span>
    </div>
  </div>
  <div
    className="cardContainer isSuccess"
    onClick={undefined}
  >
    <div
      className="title"
    >
      Unit Test
    </div>
    <div
      className="cardBody"
    >
      <span>
        <div
          className="chartStatsContainer"
        >
          <div
            className="pieChartContainer"
          >
            <div
              className="pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d"
            >
              <style
                dangerouslySetInnerHTML={
                  Object {
                    "__html": ".pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .pie-chart-slice{stroke: #fff;
stroke-width: 1;
opacity: 1;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .pie-chart-slice:hover{opacity: 0.8;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .pie-chart-label{font-family: sans-serif;
font-size: 12px;
text-anchor: middle;
fill: #000;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .bar{fill: blue;
transition: x 0.35s ease-in, y 0.35s ease-in, height 0.5s ease-in, width 0.5s ease-in;
opacity: 1;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .bar:hover{opacity: 0.8;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .line{fill: none;
stroke-width: 1.5;
opacity: 0.7;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .line:hover{opacity: 1;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .area{opacity: 0.7;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .area:hover{opacity: 1;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .dot{stroke-width: 0;
opacity: 0.85;
transition: cx 0.35s ease-in, cy 0.35s ease-in, r 0.5s ease-in;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .dot:hover{opacity: 1;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d circle.data-point{r: 4px;
opacity: 0.7;
transition: cx 0.35s ease-in, cy 0.35s ease-in;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d circle.data-point:hover{r: 6px;
opacity: 1;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d circle.tick-circle{r: 2px;
fill: lightgrey;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .x circle.tick-circle{cy: 8px;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .axis{font-family: dobra-light,Arial,sans-serif;
font-size: 9px;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .axis .label{font: 14px arial;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .axis path,.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .axis line{fill: none;
stroke-width: 1;
shape-rendering: crispEdges;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d x.axis path{display: none;
stroke: lightgrey;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .tick line{stroke: lightgrey;
stroke-width: 1;
opacity: 0.7;}.pie-chart-f564ef72ba18fa82c778ec5e18804f82c905443d .chart_text{font-size: 0.8em;
fill: #fff;}",
                  }
                }
              />
              <svg
                height={100}
                style={Object {}}
                width={100}
              >
                <g
                  id="slices-f564ef72ba18fa82c778ec5e18804f82c905443d"
                  style={Object {}}
                  transform="translate(50, 50)"
                />
                <g
                  id="labels-f564ef72ba18fa82c778ec5e18804f82c905443d"
                  style={Object {}}
                  transform="translate(50, 50)"
                />
              </svg>
            </div>
          </div>
          <div
            className="totalTests"
          >
            <span
              className="titleText"
            >
              73%
            </span>
            <span
              className="info"
            >
              tests passed
            </span>
          </div>
        </div>
        <div
          className="coverageContainer"
        >
          <span
            className="coverageCovered"
            style={
              Object {
                "opacity": 1,
                "width": "76%",
              }
            }
          />
          <span
            className="summary"
          >
            <span
              className="statsText"
            >
              76%
            </span>
            <span
              className="info"
            >
              code covered
            </span>
          </span>
        </div>
      </span>
    </div>
  </div>
  <div
    className="cardContainer isSuccess"
    onClick={undefined}
  >
    <div
      className="title"
    >
      Functional Test
    </div>
    <div
      className="cardBody"
    >
      <span>
        <div
          className="chartStatsContainer"
        >
          <div
            className="pieChartContainer"
          >
            <div
              className="pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0"
            >
              <style
                dangerouslySetInnerHTML={
                  Object {
                    "__html": ".pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .pie-chart-slice{stroke: #fff;
stroke-width: 1;
opacity: 1;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .pie-chart-slice:hover{opacity: 0.8;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .pie-chart-label{font-family: sans-serif;
font-size: 12px;
text-anchor: middle;
fill: #000;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .bar{fill: blue;
transition: x 0.35s ease-in, y 0.35s ease-in, height 0.5s ease-in, width 0.5s ease-in;
opacity: 1;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .bar:hover{opacity: 0.8;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .line{fill: none;
stroke-width: 1.5;
opacity: 0.7;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .line:hover{opacity: 1;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .area{opacity: 0.7;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .area:hover{opacity: 1;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .dot{stroke-width: 0;
opacity: 0.85;
transition: cx 0.35s ease-in, cy 0.35s ease-in, r 0.5s ease-in;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .dot:hover{opacity: 1;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 circle.data-point{r: 4px;
opacity: 0.7;
transition: cx 0.35s ease-in, cy 0.35s ease-in;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 circle.data-point:hover{r: 6px;
opacity: 1;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 circle.tick-circle{r: 2px;
fill: lightgrey;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .x circle.tick-circle{cy: 8px;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .axis{font-family: dobra-light,Arial,sans-serif;
font-size: 9px;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .axis .label{font: 14px arial;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .axis path,.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .axis line{fill: none;
stroke-width: 1;
shape-rendering: crispEdges;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 x.axis path{display: none;
stroke: lightgrey;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .tick line{stroke: lightgrey;
stroke-width: 1;
opacity: 0.7;}.pie-chart-bd29bc867a73552438ff128241eef5a0dded9ab0 .chart_text{font-size: 0.8em;
fill: #fff;}",
                  }
                }
              />
              <svg
                height={100}
                style={Object {}}
                width={100}
              >
                <g
                  id="slices-bd29bc867a73552438ff128241eef5a0dded9ab0"
                  style={Object {}}
                  transform="translate(50, 50)"
                />
                <g
                  id="labels-bd29bc867a73552438ff128241eef5a0dded9ab0"
                  style={Object {}}
                  transform="translate(50, 50)"
                />
              </svg>
            </div>
          </div>
          <div
            className="totalTests"
          >
            <span
              className="titleText"
            >
              68%
            </span>
            <span
              className="info"
            >
              tests passed
            </span>
          </div>
        </div>
        <div
          className="coverageContainer"
        >
          <span
            className="coverageCovered"
            style={
              Object {
                "opacity": 1,
                "width": "23%",
              }
            }
          />
          <span
            className="summary"
          >
            <span
              className="statsText"
            >
              23%
            </span>
            <span
              className="info"
            >
              code covered
            </span>
          </span>
        </div>
      </span>
    </div>
  </div>
  <div
    className="resultContainer"
  >
    <div
      className="resultHeader"
    >
      Result:
    </div>
    <div
      className="resultBody"
    >
      <div
        className="subTitle"
      >
        Change Rejected
      </div>
      <div
        className="reason"
      >
        Metrics Reduction
      </div>
      <button
        className="buttonContainer undefined"
        disabled={undefined}
        onClick={[Function]}
      >
        Find Issues
      </button>
    </div>
  </div>
</div>
`;
